pcss-criminal-application/src/main/java/ca/bc/gov/open/pcsscriminalapplication/service/impl/PersonnelValidatorImpl.java

```diff
@@ -0,0 +1,101 @@
+package ca.bc.gov.open.pcsscriminalapplication.service.impl;
+
+import ca.bc.gov.open.pcsscriminalapplication.service.PersonnelValidator;
+import ca.bc.gov.open.pcsscriminalapplication.utils.ValidationUtils;
+import ca.bc.gov.open.wsdl.pcss.one.GetPersonnelAvailDetailRequest;
+import ca.bc.gov.open.wsdl.pcss.one.GetPersonnelAvailabilityRequest;
+import ca.bc.gov.open.wsdl.pcss.one.GetPersonnelSearchRequest;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+@Service
+public class PersonnelValidatorImpl implements PersonnelValidator {
+
+    @Override
+    public List<String> validateGetPersonnelAvailability(GetPersonnelAvailabilityRequest getPersonnelAvailabilityRequest) {
+
+        if (getPersonnelAvailabilityRequest == null) return Collections.singletonList("Empty request is invalid");
+
+        List<String> errors = new ArrayList<>();
+
+        if (ValidationUtils.checkAgencyIdentifierTypeForErrors(getPersonnelAvailabilityRequest.getRequestAgencyIdentifierId())) {
+            errors.add("RequestAgencyIdentifierId is not valid");
+        }
+
+        if (ValidationUtils.checkSystemIdTypeForErrors(getPersonnelAvailabilityRequest.getRequestPartId())) {
+            errors.add("RequestPartId is not valid");
+        }
+
+        if (getPersonnelAvailabilityRequest.getRequestDtm() == null || ValidationUtils.checkDateTimeTypeForErrors(getPersonnelAvailabilityRequest.getRequestDtm().toString())) {
+            errors.add("RequestDtm is not valid");
+        }
+
+        if (getPersonnelAvailabilityRequest.getToDt() == null || ValidationUtils.checkDateTimeTypeForErrors(getPersonnelAvailabilityRequest.getToDt().toString())) {
+            errors.add("ToDt is not valid");
+        }
+
+        if (getPersonnelAvailabilityRequest.getFromDt() == null || ValidationUtils.checkDateTimeTypeForErrors(getPersonnelAvailabilityRequest.getFromDt().toString())) {
+            errors.add("FromDt is not valid");
+        }
```

Comment:
These are both technically DateTypes and not DateTimeTypes, so maybe we should have a different ValidationUtils function for them. Both types appear to have the same validation rules in the .xsd files though so maybe that's not neccessary.